hybrid/power_source.py

```diff
@@ -238,27 +241,35 @@ def simulate_financials(self, interconnect_kw: float, project_life: int):
         if self.system_capacity_kw <= 0:
             return
 
-        self._financial_model.FinancialParameters.analysis_period = project_life
-        self._financial_model.Lifetime.system_use_lifetime_output = 1 if project_life > 1 else 0
-        self._financial_model.Revenue.ppa_soln_mode = 1
+        self._financial_model.value('analysis_period', project_life)
+        self._financial_model.value('system_use_lifetime_output', 1 if project_life > 1 else 0)
+        self._financial_model.value('ppa_soln_mode', 1)
 
         # try to copy over system_model's generation_profile to the financial_model
-        if len(self._financial_model.SystemOutput.gen) == 1:
-            if len(self.generation_profile) == self.site.n_timesteps:
-                self._financial_model.SystemOutput.gen = self.generation_profile
+        if len(self._financial_model.value('gen')) == 1:
+            if len(self.generation_profile) == self.site.n_timesteps or \
+              len(self.generation_profile) == self.site.n_timesteps * project_life:
+                self._financial_model.value('gen', self.generation_profile)
             else:
                 raise RuntimeError(f"simulate_financials error: generation profile of len {self.site.n_timesteps} required")
 
-        if len(self._financial_model.SystemOutput.gen) == self.site.n_timesteps:
-            single_year_gen = self._financial_model.SystemOutput.gen
-            self._financial_model.SystemOutput.gen = list(single_year_gen) * project_life
-
-        if self.name != "Grid":
-            self._financial_model.SystemOutput.system_pre_curtailment_kwac = self._system_model.value("gen") * project_life
-            self._financial_model.SystemOutput.annual_energy_pre_curtailment_ac = self._system_model.value("annual_energy")
-            # TODO: Should we use the nominal capacity function here?
+        if len(self._financial_model.value('gen')) == self.site.n_timesteps:
+            self._financial_model.value('gen', self._financial_model.value('gen') * project_life)
+        self._financial_model.value('system_pre_curtailment_kwac', self._financial_model.value('gen'))
```

Comment:
Do these lines no longer require the "Grid" check? In the detailed_pv_opt branch, I still had this check, but perhaps it's no longer necessary?: https://github.com/NREL/HOPP/blob/f44626a7e6bea80b1aa2293d3d07de2a0c3be008/hybrid/power_source.py#L262

Reply:
Correct, I refactored this after finding it unnecessary.